src/parsers/srdf.hxx

```diff
@@ -280,6 +280,87 @@ namespace pinocchio
+      {
+        const std::string exception_message (filename + " does not have the right extension.");
+        throw std::invalid_argument(exception_message);
+      }
+      
+      // Open file
+      std::ifstream srdf_stream(filename.c_str());
+      if (! srdf_stream.is_open())
+      {
+        const std::string exception_message (filename + " does not seem to be a valid file.");
+        throw std::invalid_argument(exception_message);
+      }
+      
+      // Read xml stream
+      using boost::property_tree::ptree;
+      ptree pt;
+      read_xml(srdf_stream, pt);
+      
+      // Iterate over all tags directly children of robot
+      BOOST_FOREACH(const ptree::value_type & v, pt.get_child("robot"))
+      {
+        // if we encounter a tag group_state
+        if (v.first == "group_state")
+        {
+          const std::string name = v.second.get<std::string>("<xmlattr>.name");
+          if ( !model.referenceConfigurations.insert(std::make_pair(name, Model::ConfigVectorType::Zero(model.nq))).second)
+          {
+            //  Element already present...
+            if (verbose) std::cout << "The reference configuration "
+                                   << name << " has been defined multiple times. "
+                                   <<"Only the last instance of "<<name<<" is being used."
+                                   <<std::endl;
+          }
+          typename Model::ConfigVectorType& ref_config = model.referenceConfigurations.find(name)->second;
+          
+          // Iterate over all the joint tags
+          BOOST_FOREACH(const ptree::value_type & joint_tag, v.second)
+          {
+            if (joint_tag.first == "joint")
+            {
+              std::string joint_name = joint_tag.second.get<std::string>("<xmlattr>.name");
+              typename Model::JointIndex joint_id = model.getJointId(joint_name);
+
+              // Search in model the joint and its config id
+              if (joint_id != model.joints.size()) // != model.njoints
+              {
+                const JointModel & joint = model.joints[joint_id];
+                typename Model::ConfigVectorType joint_config(joint.nq());
+                const std::string joint_val = joint_tag.second.get<std::string>("<xmlattr>.value");
+                std::istringstream config_string(joint_val);
+                std::vector<double> config_vec((std::istream_iterator<double>(config_string)), std::istream_iterator<double>());
+                joint_config = Eigen::Map<Eigen::VectorXd>(config_vec.data(), config_vec.size());
+                ref_config.segment(joint.idx_q(),joint.nq()) = joint_config;
+                if (verbose)
+                {
+                  std::cout << "(" << joint_name << " , " << joint_config.transpose() << ")" << std::endl;
+                }
+              }
+              else
+              {
+                if (verbose) std::cout << "The Joint " << joint_name << " was not found in model" << std::endl;
```

Comment:
I think the handling of error messages is not very convenient. Either you enable logs and you get a whole bunch of useless message, with maybe an error in between, or you don't enable logs and you don't get any error messages.

I think would be smarter to put this in std::cerr, even when verbose if false.

This remark is related to https://github.com/stack-of-tasks/pinocchio/issues/375

Reply:
Thanks @jmirabel
Error handling and logging is a much bigger issue, I mainly followed the existing practice when loading the srdf. Just editing this one error might not be sufficient for what is eventually desired.
I'll let the gurus take care of this :)